### PR TITLE
Update yggdrasil

### PR DESCRIPTION
**Pull request by wvffle**
As the author of node-yggdrasil doesn't seem to accept new PRs, I've created a fork which fixes phin error (thanks to @rom1504) and adds more meaningful messages for a couple of CloudFlare errors as proposed in https://github.com/PrismarineJS/mineflayer/issues/880#issuecomment-549127642

### CODE REVIEW

**Comment by wvffle** [comment]
Also, as it may seem not obvious here is @waff/yggdrasil: https://github.com/wvffle/yggdrasil

**Comment by IdanHo** [comment]
I've been using wvffle's version of node-yggdrasil for the past couple of weeks continously due to phin errors I've experienced with the older version and i can confirm it's working as expected.

**Comment by plexigras** [comment]
this sounds like a security risk

**Comment by wvffle** [comment]
You mean using a fork? If we're going this way, using any library that's outside of the organization is a security risk. That's why we use a specific version of a library (which code we've reviewed) and not blindly the latest version.

If you mean that these changes seem like a security risk then I don't get it, please explain.

**Comment by rom1504** [comment]
Didn't have time to look at this yet, but I'll have a look today

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "standard": "^14.0.0"
   },
   "dependencies": {
+    "@waff/yggdrasil": "^1.1.4",
```
Comment (plexigras) [inline package.json]: > That's why we use a specific version of a library (which code we've reviewed) and not blindly the latest version.

shouldn't it be `1.1.4` rather then `^1.1.4 := >=1.1.4 <2.0.0`

Comment (rom1504) [reply package.json]: no, that's how semver works

**Comment by rom1504** [comment]
I think it's a shame to duplicate the repo.
Could you try to contact the author ? we could move his repo to prismarinejs if he doesn't want to maintain it anymore.

**Comment by wvffle** [comment]
In my opinion it's not a shame. That's how open source works :)
No, I haven't contacted the author but it seems like he does not care for the issues and PRs. Tomorrow I'll send him a message asking if he's going to maintain it or not.

**Comment by rom1504** [comment]
My point is it's better to reuse an existing name than to create a new one.
This one has already a bunch of stars, a bunch of people using it and that
could potentially contribute.
When you create a new name out of nothing, you need to recreate the
community.
So it would be best to either make him look at the pr, either become
maintainer of his repo, either move the repo to prismarinejs organization
so we can have several people looking at it.
I think moving the repo to prismarinejs is probably the best option.

**Comment by wvffle** [comment]
@zekesonxx Are you going to support https://github.com/zekesonxx/node-yggdrasil? There are some unresolved issues in your library. I've made a fork of it but we're discussing making a PR to yggdrasil or moving it to the @PrismarineJS where we could work on it if you are not going maintain it anymore :)

Let us know :)

**Comment by zekesonxx** [comment]
No, I don't have any plans to deal with it. In all honesty I thought I had handed control over to y'all a while ago, since you're clearly going to do much more with it than I'm ever going to.

**Comment by zekesonxx** [comment]
If I can transfer control of it to prismarine, that would be great, let it get some improvements and thrive.

**Comment by rom1504** [comment]
@zekesonxx ok that's great ! I added you to the prismarinejs org. You just have to accept the invite and then you should be able to move the GitHub project to the organization.
Can you also add wvffle and me on npm ?

**Comment by rom1504** [comment]
@zekesonxx did you manage to move the repo ?

**Comment by zekesonxx** [comment]
https://github.com/PrismarineJS/node-yggdrasil

**Comment by zekesonxx** [comment]
I've also invited rom1504 and wvffle to be maintainers on the npm repo.

**Comment by rom1504** [comment]
thanks @zekesonxx ; I didn't seem to received any invitation for npm though, did you sent it to the right user ?
my npm user is https://www.npmjs.com/~rom1504

@wvffle could you open a PR with the changes you made on node-yggdrasil ?

**Comment by zekesonxx** [comment]
@rom1504 yep, it's even listed in your npm profile.

**Comment by rom1504** [comment]
ah right, I see it now

**Pull request merged**